From 58da1aaeb62e3fdf91bbf2b83cac9f3add0b5246 Mon Sep 17 00:00:00 2001
From: denglei <denglei@eswincomputing.com>
Date: Thu, 24 Apr 2025 15:43:26 +0800
Subject: [PATCH 489/654] fix:Resolve the issue of NPU reset.

Changelogs:
	Running the model after NPU reset causes SMMU SYNC timeout.

Signed-off-by: denglei <denglei@eswincomputing.com>
---
 drivers/soc/eswin/ai_driver/npu/nvdla_hw.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/eswin/ai_driver/npu/nvdla_hw.c b/drivers/soc/eswin/ai_driver/npu/nvdla_hw.c
index 1addd530c1f7c..6e2f83bd8b97a 100644
--- a/drivers/soc/eswin/ai_driver/npu/nvdla_hw.c
+++ b/drivers/soc/eswin/ai_driver/npu/nvdla_hw.c
@@ -516,6 +516,8 @@ static ssize_t store_reset_hand(struct device *d, struct device_attribute *attr,
 
 			engine->tiktok = 0;
 
+			edma_free(nvdla_dev);
+
 			ret = npu_hardware_reset(nvdla_dev);
 			if (ret) {
 				dla_error("hardware reset err, ret=%d.\n", ret);
@@ -575,11 +577,16 @@ static ssize_t store_reset_hand(struct device *d, struct device_attribute *attr,
 				dla_error("e31 not bootup, error.\n");
 				goto exit;
 			}
+
+			ret = edma_init(nvdla_dev);
+			if (ret) {
+				dla_error("edma_init fail\n");
+				goto exit;
+			}
 			engine->engine_is_alive = true;
 		} else {
 			dla_error("err:input err.\n");
 		}
-		goto exit;
 	}
 
 exit:
-- 
2.51.0

